From fc0623de81ecabf7b3ee7d4bae9b581c10af2954 Mon Sep 17 00:00:00 2001
From: Christof Kaufmann <christof.kaufmann@hs-bochum.de>
Date: Wed, 19 Apr 2023 01:42:34 +0200
Subject: [PATCH] Fix compatibility version mismatch with shimmy

---
 tests/envs/test_compatibility.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/envs/test_compatibility.py b/tests/envs/test_compatibility.py
index f23ee3cd..3c4e0954 100644
--- a/tests/envs/test_compatibility.py
+++ b/tests/envs/test_compatibility.py
@@ -177,5 +177,5 @@ def test_shimmy_gym_compatibility():
         ):
             gymnasium.make("GymV26Environment-v0", env_id="CartPole-v1")
     else:
-        gymnasium.make("GymV22Environment-v0", env_id="CartPole-v1")
+        gymnasium.make("GymV21Environment-v0", env_id="CartPole-v1")
         gymnasium.make("GymV26Environment-v0", env_id="CartPole-v1")
-- 
2.40.0

