diff --git a/gymnasium/spaces/multi_discrete.py b/gymnasium/spaces/multi_discrete.py
index 09f930ff8..caf488764 100644
--- a/gymnasium/spaces/multi_discrete.py
+++ b/gymnasium/spaces/multi_discrete.py
@@ -172,7 +172,7 @@ def from_jsonable(
         self, sample_n: list[Sequence[int]]
     ) -> list[NDArray[np.integer[Any]]]:
         """Convert a JSONable data type to a batch of samples from this space."""
-        return [np.array(sample) for sample in sample_n]
+        return [np.array(sample, dtype=np.int64) for sample in sample_n]
 
     def __repr__(self):
         """Gives a string representation of this space."""
